B 22469 int part3

r-mettler opened the pull request:
## [B-22469](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1063924)

## Summary

Discussion was had and it was decided to put the City State ZIP and County fields back underneath the Location Lookup for the City Finder. This PR adds the fields back to the UI as well as adds back tests for those fields (a majority of the changes here). Also fixed a bug with the Editing of PPM shipment addresses during PPM Closeout. When selecting a new location from the search results Formik was not re-validating, the fix for this is in this PR. 

Is there anything you would like reviewers to give additional scrutiny?

Test various address lookup across all apps, customer, office and prime sim.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Create a customer and PPM shipment, verify the Address components display the City State ZIP and County fields beneath the search box and that they are populated when selecting a search result and Formik validates again and enables the save button if its the last thing to be changed on the page. 

Additionally during PPM closeout (after clicking Review Documents as a SC) click the pencil icon for at least one address and search for and select a new location and make sure you can save the change.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### Database

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots

antgmann reviewed (commented):
I've got a couple of teeny tiny nitpicks - Happo is showing some storybook components with initial values don't have counties (First image happo, second image includes a county I added on my local)
<img width="922" alt="image" src="https://github.com/user-attachments/assets/8f8c5bb6-52c8-4c44-99d9-a9814ee81c7f" />
<img width="971" alt="image" src="https://github.com/user-attachments/assets/4fc77a7c-62a1-4385-8be0-df7a1611f115" />

antgmann commented on src/components/Office/PPM/PPMHeaderSummary/EditPPMHeaderSummaryModal.jsx:
```diff
@@ -92,7 +92,8 @@ const EditPPMHeaderSummaryModal = ({ sectionType, sectionInfo, onClose, onSubmit
           >
-            {({ isValid, handleChange, setFieldTouched, values, ...formikProps }) => {
+            {({ isValid, handleChange, values, ...formikProps }) => {
+              const setfieldtouched = formikProps.setFieldTouched;
```
Could this be const setFieldTouched?

r-mettler replied:
yes it could, but its not 😄 
I'll change it while making changes for storybook.

r-mettler replied:
Removed the const as it was not needed, just assigned using the ```formik.setFieldTouched``` directly

antgmann commented:
...and that's it! Everything else looks good on my local, and I'll attach the ANDI screenshot to the backlog. 

There are some failing playwright tests - but from what I can tell, none of them are a result of the changes on the PR

traskowskycaci reviewed (dismissed):
Ran through the PPM closeout flow (without StrictMode!) and all looked good. Did check editing addresses as SC during the Review Documents step and that checkout out as well.

r-mettler commented:
> ...and that's it! Everything else looks good on my local, and I'll attach the ANDI screenshot to the backlog.
> 
> There are some failing playwright tests - but from what I can tell, none of them are a result of the changes on the PR

Yeah the failing playwright tests are existing test failures. I have tried to fix as many of them as I can but its a never ending battle.

r-mettler commented:
> I've got a couple of teeny tiny nitpicks - Happo is showing some storybook components with initial values don't have counties (First image happo, second image includes a county I added on my local) 

Updated storybook to have the county. Also fixed the ```LocationSearchBox``` story for the City Finder since it was not working properly (not displaying the location in the box after selection).

antgmann reviewed (approved):
Thanks for the fixes!

cameroncaci reviewed (approved):
Backend code is good
